modal/partial_function.py

```diff
@@ -53,13 +53,24 @@ def __init__(
         self.wrapped = False  # Make sure that this was converted into a FunctionHandle
 
     def __get__(self, obj, objtype=None) -> _Function:
-        # This only happens inside user methods when they refer to other methods
         k = self.raw_f.__name__
         if obj:  # Cls().fun
-            function = getattr(obj, "_modal_functions")[k]
-        else:  # Cls.fun
-            function = getattr(objtype, "_modal_functions")[k]
-        return function
+            if hasattr(obj, "_modal_functions"):
+                # This happens inside "local" user methods when they refer to other methods,
+                # e.g. Foo().parent_method() doing self.local.other_method()
+                return getattr(obj, "_modal_functions")[k]
```

Comment:
This feels brittle, although I don't have a better suggestion right now.

Reply:
I agree, and I'm not even sure exactly when this triggers vs the `__getattr__` on `_Obj` tbh...

Hopefully we can clean this up (it's all preexisting code, but I removed the part of the code that sets this attribute on **the class** in this patch, so now it's only being set on the instance of the user class that's being bound to the function in the container entrypoint)

Reply:
There's a slightly more complex solution where we use the cls the user provided, or construct a class dynamically in the case of lookups. That would let us get rid of getattr "hack"